tests/expand/h014-in.html

```diff
@@ -0,0 +1,14 @@
+<html>
+  <head>
+    <script type="application/ld+json">
+    <!--
+    {
+      "@context": {
+        "foo": {"@id": "http://example.com/foo", "@container": "@list"}
+      },
+      "foo": [{"@value": "<!-- -->"}]
```

Comment:
Certainly, HTML embedded in HTML contained in an HTML script would be one such use. Also, various HTML validators complain when script elements contain things that it doesn't expect, and placing comments around the entire JSON-LD block is one solution for this.

Intermingling without escapes is not supported, to be intermingled, they must be escaped. This could be using `<\!-- --\>` or `&lt;!-- --&gt;`. Easiest thing is to comment out the whole block, and deal specifically with any embedded XML comments rather than entity encode the contents; either would work, though.